zk_toolbox/crates/zk_inception/src/commands/chain/register_chain.rs

```diff
@@ -0,0 +1,98 @@
+use anyhow::Context;
+use common::{
+    config::global_config,
+    forge::{Forge, ForgeScriptArgs},
+    logger,
+    spinner::Spinner,
+};
+use config::{
+    forge_interface::{
+        register_chain::{input::RegisterChainL1Config, output::RegisterChainOutput},
+        script_params::REGISTER_CHAIN_SCRIPT_PARAMS,
+    },
+    traits::{ReadConfig, SaveConfig, SaveConfigWithBasePath},
+    ChainConfig, ContractsConfig, EcosystemConfig,
+};
+use xshell::Shell;
+
+use crate::{
+    messages::{
+        MSG_CHAIN_NOT_INITIALIZED, MSG_CHAIN_REGISTERED, MSG_L1_SECRETS_MUST_BE_PRESENTED,
+        MSG_REGISTERING_CHAIN_SPINNER,
+    },
+    utils::forge::{check_the_balance, fill_forge_private_key},
+};
+
+pub async fn run(args: ForgeScriptArgs, shell: &Shell) -> anyhow::Result<()> {
```

Comment:
TBH, i'm thinking about moving it to the ecosystem subcommand, because it's all happening on the ecosystem level

Reply:
will do, just updated with genesis chunked